### ISSUE TITLE
Mark charms as AR-changing and include/exclude them from AR queries

### ISSUE THREAD

**Issue by AardWolf** [DevJackSmith/mh-hunt-helper#30]
For example, warpath charms greatly change the AR in the area. So when you look up something like caravan guard he shows very low although he's much higher if you don't have any warpath charm armed.

Not sure how you'd implement it - could add a flag to the hunt if an AR charm was used (not sure if the specific one is important), could change the stage to something like "Stage 1 - Archer charm".

**Comment by DevJackSmith** [comment]
i am currently recording charms for each hunt. I think what you are trying to say is "add AR affecting charms to attraction rate tool". Good suggestion, i was planning on it. Just need time to do it. (If i misunderstood, please clarify, otherwise I'll assume this is what you meant).

**Comment by AardWolf** [comment]
Altering this to suggest another implementation (which I may get to add a PR for):

Using Muridae as an example, add two stages - Artisan and Not Artisan. Once these are recording run a query similar to the follow to retroactively fix the hunts:

```
INSERT INTO HUNT_STAGES SELECT HUNT_ID, (artisan_stage_id) FROM HUNTS WHERE LOCATION_ID = muridae and charm_id = artisan_charm;
INSERT INTO HUNT_STAGES SELECT HUNT_ID, (non_artisan_stage_id) FROM HUNTS WHERE LOCATION_ID = muridae and charm_id != artisan_charm;
```

**Comment by tsitu** [comment]
I thought we decided not to separate AR-affecting charms into distinct stages? Thoughts, @DevJackSmith?

**Comment by DevJackSmith** [comment]
I agree with Tsitu, we don't want to separate AR charms into their own stages. Reason being that would multiply our current stages by number of AR charms. Too many stages. If we do it just for few areas, we are not being consistent across the board. I suggest we mark them as AR changing charms. For example add another column in the charms table that is named ar-changing or something. But that would be bad too since some charms only change ar in some areas and not others as far as i can tell. So any other ideas? (we want to stay consistent everywhere)

**Comment by DevJackSmith** [comment]
we can implement logic and mark hunts that have ar changing charm in a correct location maybe? hunts table is huge though, so i'd want to separate charms out of it then with a translation table (hunt_charm) that has 3 columns (hunt_id, charm_id, ar-affecting)

**Comment by DevJackSmith** [comment]
the logic can pull data from a table which has 1. charm_id 2. locations where it's ar-affecting (with a wildcard keyword 'all' so we don't have to list all locations)

**Comment by AardWolf** [comment]
Some areas have charms specifically to change ar. Softserve and artisan come to mind. They'd make sense as stages in my mind.

Something like brain and super brain change ar but more generically and don't necessarily need stages. 

One implementation would be adding a table with charm id and location id for locations where ar is specifically changed by charms identified. That's something that could be thrown together for those of us who download the data set. 

If we wanted to make it available generally as a page it could be a new one that has ar by location and lets the user filter by stage, cheese, and/or any ar-changing charms marked for that area.
